### PR TITLE
add basic compiler selection telemetry

### PR DESCRIPTION
**Pull request by browntarik**

### CODE REVIEW

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -971,6 +971,8 @@ export class DefaultClient implements Client {
         paths.push(localize("installCompiler.string", "Help me install a compiler"));
         paths.push(localize("noConfig.string", "Do not configure a compiler (not recommended)"));
         const index: number = await this.showSelectDefaultCompiler(paths);
+        const action: string = index === -1 ? 'escaped' : index === paths.length - 1 ? 'disable' : index === paths.length - 2 ? 'help' : index === paths.length - 3 ? 'browse' : 'select compiler';
```
Comment (michelleangela) [inline Extension/src/LanguageServer/client.ts]: For readability, can this be replaced with switch/cases or if/else? Nested trinary are hard to follow.

**Review by michelleangela — CHANGES_REQUESTED** [review]

File: Extension/src/LanguageServer/client.ts [hunk 2]
```diff
@@ -971,6 +971,8 @@ export class DefaultClient implements Client {
...
+        telemetry.logLanguageServerEvent('compilerSelection', { action });
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/client.ts:992]: What about escape after browse? Do we want telemetry on that?

Comment (bobbrow) [reply Extension/src/LanguageServer/client.ts:992]: I think we can address that in a follow up PR.  There are a few telemetry requests in the spec that weren't implemented.  I'd like to get some basic stuff for 1.14.2 without missing our deadline

File: Extension/src/LanguageServer/client.ts (hunk 1 above)
Comment (bobbrow) [reply Extension/src/LanguageServer/client.ts]: Let's put a note to clean it up for 1.14.3.  I mostly just don't want to jeapordize today's pre-release.

**Review by bobbrow — DISMISSED** [review]

**Review by sean-mcmanus — DISMISSED** [review]

**Review by michelleangela — DISMISSED** [review]

**Review by michelleangela — APPROVED** [review]

**Review by sean-mcmanus — APPROVED** [review]
